tests/components/zeroconf/test_init.py

```diff
@@ -123,14 +131,19 @@ async def test_homekit_match_full(hass, mock_zeroconf):
     assert mock_config_flow.mock_calls[0][1][0] == "hue"
 
 
-async def test_info_from_service_non_utf8(hass):
-    """Test info_from_service handles non UTF-8 property values correctly."""
+async def test_info_from_service_non_utf8(hass, caplog):
+    """Test info_from_service handles non UTF-8 property keys and values correctly."""
     service_type = "_test._tcp.local."
-    info = zeroconf.info_from_service(
-        get_service_info_mock(service_type, f"test.{service_type}")
-    )
+    service_info_mock = get_service_info_mock(service_type, f"test.{service_type}")
+    info = zeroconf.info_from_service(service_info_mock)
     raw_info = info["properties"].pop("_raw", False)
     assert raw_info
+    assert (
```

Comment:
To make this assertion more future proof, you could change the assertion to something like:
```python
    assert (
        len(
            [
                log
                for log in caplog.records
                if log.name == "homeassistant.components.zeroconf"
                and log.levelname == "DEBUG"
            ]
        )
        == 1
    )
```
this way you are not dependent on the actual log message

Reply:
Or perhaps checking the debug log is unnecessary as we're already validating the data itself. I think I'll remove the caplog check.